docs/test_suite.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Test Suite
+keywords: test suite
+sidebar: mydoc_sidebar
+permalink: test_suite.html
+toc: false
+summary: In this page, we will discuss the Hudi Test suite to perform end to end tests
+
+This page describes in detail how to run end to end tests on a hudi dataset that helps in improving our confidence 
+in a release as well as perform large scale performance benchmarks.  
+
+### Objectives
+
+1. Test with different versions of core libraries and components such as `hdfs`, `parquet`, `spark`, 
+`hive` and `avro`.
+2. Generate different types of workloads across different dimensions such as `payload size`, `number of updates`, 
+`number of inserts`, `number of partitions`
+3. Perform multiple types of operations such as `insert`, `bulk_insert`, `upsert`, `compact`, `query`
+4. Support custom post process actions and validations
+
+### High Level Design
+
+{% include image.html file="hudi_test_suite_design.png" alt="hudi_test_suite_design.png" %}
```

Comment:
Some terminology like : Operation and Action is defined in the image. Trying to understand - Is this is still applicable after your refactor ? 

Also, Can you elaborate on the terminologies in this readme.

Reply:
The image is old, let me fix it.